cmd/restic/cmd_backup.go

```diff
@@ -373,19 +420,34 @@ func collectTargets(opts BackupOptions, args []string) (targets []string, err er
 			if len(expanded) == 0 {
 				Warnf("pattern %q does not match any files, skipping\n", line)
 			}
-			lines = append(lines, expanded...)
+			targets = append(targets, expanded...)
+		}
+	}
+
+	for _, file := range opts.FilesFromVerbatim {
+		fromfile, err := readPatternsFromFile(file, false)
+		if err != nil {
+			return nil, err
+		}
+		targets = append(targets, fromfile...)
+	}
+
+	for _, file := range opts.FilesFromRaw {
+		fromfile, err := readFilenamesFromFileRaw(file)
+		if err != nil {
+			return nil, err
 		}
+		fmt.Println(file, fromfile)
+		targets = append(targets, fromfile...)
 	}
 
-	// merge files from files-from into normal args so we can reuse the normal
-	// args checks and have the ability to use both files-from and args at the
-	// same time
-	args = append(args, lines...)
+	// Merge args into files-from so we can reuse the normal args checks
+	// and have the ability to use both files-from and args at the same time.
+	targets = append(targets, args...)
```

Comment:
The line *below this one I'm commenting on* should probably be checking `len(targets)` instead?

Reply:
Done.